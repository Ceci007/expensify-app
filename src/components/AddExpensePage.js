import React from 'react';

const AddExpensePage = () => (
  <div>
    This is from the add expense component.
  </div>
)

export default AddExpensePage;