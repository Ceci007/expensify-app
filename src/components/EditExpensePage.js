import React from 'react';

const EditExpensePage = () => (
  <div>
    This is from the edit expense component.
  </div>
)

export default EditExpensePage;