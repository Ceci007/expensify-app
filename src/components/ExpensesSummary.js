import React from 'react';
import {Link} from 'react-router-dom';
import { connect } from 'react-redux';
import numeral from 'numeral';
import getVisiblesExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({expensesCount, expensesLength, expensesTotal}) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal).format('$0,0.00');
    const hiddenExpensesCount = expensesLength !== expensesCount ? expensesLength - expensesCount : 0;

    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">
                    Viewing <span>{expensesCount}</span> {expenseWord} totalling <span>{formattedExpensesTotal}</span>
                    {hiddenExpensesCount > 0 && <div className="page-header__note">{hiddenExpensesCount} expenses added, but they are hidden, try clearing the date range</div>}
                </h1>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = getVisiblesExpenses(state.expenses, state.filters);
    const total = getExpensesTotal(visibleExpenses) / 100;

    return {
        expensesCount: visibleExpenses.length,
        expensesLength: state.expenses.length,
        expensesTotal: total,
    };
};

export default connect(mapStateToProps)(ExpensesSummary);