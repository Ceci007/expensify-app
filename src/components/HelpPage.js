import React from 'react';

const HelpPage = () => (
  <div>
    This is from the help component.
  </div>
)

export default HelpPage;
