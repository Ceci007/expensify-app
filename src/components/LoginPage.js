import React from 'react';
import { connect } from 'react-redux';
import {
    startLoginWithGoogle
} from '../actions/auth';

export const LoginPage = ({ startLoginWithGoogle }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify</h1>
            <p>It's time to get your expenses under control</p>
            <button className="button button--google" onClick={startLoginWithGoogle}>Login with Google</button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLoginWithGoogle: () => dispatch(startLoginWithGoogle()),
    startLoginWithFacebook: () => dispatch(startLoginWithFacebook()),
    startLoginWithTwitter: () => dispatch(startLoginWithTwitter()),
    startLoginWithGithub: () => dispatch(startLoginWithGithub()),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);