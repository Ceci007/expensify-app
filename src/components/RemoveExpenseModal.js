import React, { Component } from 'react';
import Modal from 'react-modal';
import { connect } from 'react-redux';
import { history } from '../routers/AppRouter';
import { startRemoveExpense } from '../actions/expenses';
import { cancelRemoveExpense } from '../actions/expense';

export class RemoveExpenseModal extends Component {
    handleRemoveExpense = () => {
        this.props.startRemoveExpense(this.props.expense);
        this.props.cancelRemoveExpense();
        this.props.history.push('/');
    };

    render() {
        return (
            <Modal
                isOpen={this.props.showModal}
                onRequestClose={this.props.cancelRemoveExpense}
                contentLabel="Remove expense"
                ariaHideApp={false}
                closeTimeoutMS={200}
                className="modal">
                <h3 className="modal__title">Remove expense</h3>
                Are you sure you want to remove <strong>{this.props.expense.description}</strong> expense?
                <div className="modal__actions">
                    <button className="button button--danger" id="remove" onClick={this.handleRemoveExpense}>Remove
                    </button>
                    <button className="button button--secondary" onClick={this.props.cancelRemoveExpense}>Cancel
                    </button>
                </div>
            </Modal>
        );
    }
}

const mapStateToProps = (state) => ({
    expense: state.expense,
    showModal: !!state.expense.id,
    history: history,
});

const mapDispatchToProps = (dispatch) => ({
    startRemoveExpense: (data) => {
        dispatch(startRemoveExpense(data));
    },
    cancelRemoveExpense: () => {
        dispatch(cancelRemoveExpense());
    },
});

export default connect(mapStateToProps, mapDispatchToProps)(RemoveExpenseModal);