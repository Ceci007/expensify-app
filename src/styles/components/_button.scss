.button {
  background: $blue;
  border: none;
  color: white;
  display: inline-block;
  padding: $s-size $l-size;
  font-weight: 300;
  font-size: $font-size-large;
  line-height: 1;
  text-decoration: none;
}

.button--link {
  background: none;
}

.button--secondary {
  background: #888;
}

.button--danger {
  background: $danger-red;
  margin-right: 2rem;

  @media (max-width: $desktop-breakpoint) {
    margin-right: 0;
    margin-bottom: 15px;
  }
}

.two-items {
  display: flex;
  justify-content: space-between;
}