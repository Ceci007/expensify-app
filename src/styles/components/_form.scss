.form {
  display: flex;
  flex-direction: column;

  > * {
    margin-bottom: $m-size;
  }
}

.form__error {
  margin: 0 0 $m-size 0;
  font-style: italic;
  background: #ef9a9a;
  color: $danger-red;
  padding: 10px;
  border: 2px solid $danger-red;
}