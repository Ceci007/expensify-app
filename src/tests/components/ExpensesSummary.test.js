import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('Should render correctly ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={235}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
});

test('Should render correctly ExpensesSummary with multiples expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={23} expensesTotal={23521321}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
});