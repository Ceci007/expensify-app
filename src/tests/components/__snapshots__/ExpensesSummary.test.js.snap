// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly ExpensesSummary with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $235.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <[object Object]
        className="button"
        to="/create"
      >
        Add Expense
      </[object Object]>
    </div>
  </div>
</div>
`;

exports[`Should render correctly ExpensesSummary with multiples expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        23
      </span>
       
      expenses
       totalling 
      <span>
        $23,521,321.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <[object Object]
        className="button"
        to="/create"
      >
        Add Expense
      </[object Object]>
    </div>
  </div>
</div>
`;
